client/src/components/CheckBox/CheckBox.unit-test.tsx

```diff
@@ -0,0 +1,78 @@
+import { render } from "@testing-library/react";
+import React from "react";
+
+import { is_a11y_mode } from "src/core/injected_build_constants";
+
+import { with_console_error_silenced } from "src/testing_utils";
+
+import { CheckBox } from "./CheckBox";
+
+const non_solid_checkbox_no_onclick = <CheckBox id={"id"} label={"label"} />;
+const solid_checkbox_no_onclick = (
+  <CheckBox
+    id={"id"}
+    label={"label"}
+    isSolidBox={true}
+    active={true}
+    color={"#000000"}
+  />
+);
+const solid_checkbox_with_onclick = (
+  <CheckBox
+    id={"id"}
+    label={"sample label"}
+    isSolidBox={true}
+    active={false}
+    onClick={(_id: string) => console.log("clicked!")}
+  />
+);
+
+describe("CheckBox", () => {
+  it('Throws an error when a solid checkbox is missing "active" and "onclick"', () => {
+    with_console_error_silenced(() => {
+      expect(() => render(non_solid_checkbox_no_onclick)).toThrow();
+    });
+  });
+
+  it("Renders a simple, unstyled CheckBox in a11y mode", () => {
+    if (is_a11y_mode) {
+      render(solid_checkbox_no_onclick);
+      expect(
+        document.querySelector("div.checkbox label input[type='checkbox']")
+      ).toBeTruthy();
+    }
+  });
+
+  it("Displays the given label when an onclick is provided", () => {
+    render(solid_checkbox_with_onclick);
+    const checkbox_span = document.querySelector(
+      is_a11y_mode ? "div.checkbox label input" : "div span.link-styled"
+    );
+    expect(checkbox_span && checkbox_span.textContent).toEqual("sample label");
```

Comment:
Best to try and avoid this sort of reliance on detailed implementation details, makes the tests less useful when refactoring. Keep React Testing Library in mind, it's a bit indirect some times but it focus on selecting by things that would be user facing, which is a good approach to keep the tests implementation-detail-agnostic for the most part.

The React Testing Library way would probably be to import `screen` and use `screen.getByText("sample label")`. E.g. `expect(screen.getByText("sample label")).toBeTruthy()` (or maybe something stricter like `.toBeVisible()`, still fresh enough to their approach that I've not built up go-to patterns for this my self).